device.go

```diff
@@ -284,42 +292,37 @@ func updateSpecDeviceList(device pb.Device, spec *pb.Spec) error {
 	}).Info("handling block device")
 
 	// Update the spec
-	for idx, d := range spec.Linux.Devices {
-		if d.Path == device.ContainerPath {
-			hostMajor := spec.Linux.Devices[idx].Major
-			hostMinor := spec.Linux.Devices[idx].Minor
-			agentLog.WithFields(logrus.Fields{
-				"device-path":        device.VmPath,
-				"host-device-major":  hostMajor,
-				"host-device-minor":  hostMinor,
-				"guest-device-major": major,
-				"guest-device-minor": minor,
-			}).Info("updating block device major/minor into the spec")
-
-			spec.Linux.Devices[idx].Major = major
-			spec.Linux.Devices[idx].Minor = minor
-
-			// there is no resource to update
-			if spec.Linux == nil || spec.Linux.Resources == nil {
-				return nil
-			}
+	idxData, ok := devIdx[device.ContainerPath]
+	if !ok {
+		return grpcStatus.Errorf(codes.Internal,
+			"Should have found a matching device %s in the spec",
+			device.ContainerPath)
+	}
 
-			// Resources must be updated since they are used to identify the
-			// device in the devices cgroup.
-			for idxRsrc, dRsrc := range spec.Linux.Resources.Devices {
-				if dRsrc.Major == hostMajor && dRsrc.Minor == hostMinor {
-					spec.Linux.Resources.Devices[idxRsrc].Major = major
-					spec.Linux.Resources.Devices[idxRsrc].Minor = minor
-				}
-			}
+	agentLog.WithFields(logrus.Fields{
+		"device-path":        device.VmPath,
+		"host-device-major":  spec.Linux.Devices[idxData.idx].Major,
+		"host-device-minor":  spec.Linux.Devices[idxData.idx].Minor,
+		"guest-device-major": major,
+		"guest-device-minor": minor,
+	}).Info("updating block device major/minor into the spec")
 
-			return nil
-		}
+	spec.Linux.Devices[idxData.idx].Major = major
+	spec.Linux.Devices[idxData.idx].Minor = minor
+
+	// there is no resource to update
+	if spec.Linux == nil || spec.Linux.Resources == nil {
```

Comment:
I would add a condition on `resourceIdx` here, since your loop below depends on it. Given how `devIdx` is constructed, I think it is the only necessary condition, but I prefer to replicate the same conditions used in `makeDevIndex`

Reply:
Actually, I think it makes sense to drop those conditions entirely - they're redundant with the checks in makeDevIndex.

We don't need a condition on resourceIdx: makeDevIndex creates an empty slice (rather than nil) if there is nothing matching.  That's deliberate so that we don't need another special case here.